### PR TITLE
Fix #242

### LINKED ISSUE #242: Event handler getters should return a single function
Right now, event handler's (like `onopen`, and `onerror`) getters should return a single function instead of the present behavior where they return an array. I haven't gone through the spec but that's what the browser does. If that sounds correct I'm happy to send a PR.

### PR DESCRIPTION
**Pull request by ritave**
fixes #242

### CODE REVIEW

File: src/websocket.js [hunk 1]
```diff
@@ -94,39 +99,67 @@ class WebSocket extends EventTarget {
...
   get onerror() {
-    return this.listeners.error;
+    return this.onerrorCb;
   }
 
   set onopen(listener) {
-    delete this.listeners.open;
-    this.addEventListener('open', listener);
+    if (listener === null) {
+      if (this.onopenCb !== null) {
```
Comment (Atrue) [inline src/websocket.js:119]: you can omit this additional check. `removeEventListener` takes care of it

File: src/websocket.js [hunk 2]
```diff
@@ -94,39 +99,67 @@ class WebSocket extends EventTarget {
...
+        this.removeEventListener('open', this.onopenCb);
```
Comment (Atrue) [inline src/websocket.js:120]: you need *always* remove the previous event listener, not only when the argument is null

File: src/websocket.js [hunk 3]
```diff
@@ -94,39 +99,67 @@ class WebSocket extends EventTarget {
...
+      }
+    } else {
+      this.onopen = null;
```
Comment (Atrue) [inline src/websocket.js:123]: infinity recursion

File: src/websocket.js [hunk 4]
```diff
@@ -21,6 +21,11 @@ class WebSocket extends EventTarget {
   constructor(url, protocols) {
     super();
 
+    this.onopenCb = null;
```
Comment (Atrue) [inline src/websocket.js:24]: please use underscore properties to mark them as private

**Review by Atrue — CHANGES_REQUESTED** [review]

File: src/websocket.js (hunk 4 above)
Comment (ritave) [reply src/websocket.js:24]: The linter is not happy about that. I can edit the rules of the linter

File: src/websocket.js (hunk 3 above)
Comment (ritave) [reply src/websocket.js:123]: It's not infinite recursion at all. If the callback is null, it never takes that branch

File: src/websocket.js (hunk 2 above)
Comment (ritave) [reply src/websocket.js:120]: It removes the previous listener below as well using `this.onopen = null`

**Review by Atrue — APPROVED** [review]

**Pull request merged**